Mathlib/CategoryTheory/ChosenFiniteProducts.lean

```diff
@@ -757,18 +766,30 @@ lemma μ_comp [(F ⋙ G).Monoidal] (X Y : C) : μ (F ⋙ G) X Y = μ G _ _ ≫ G
 variable [PreservesFiniteProducts F]
 
 lemma ε_of_chosenFiniteProducts : ε F = (preservesTerminalIso F).inv := by
-  change (εIso F).symm.inv = _; congr; ext; simpa using η_of_chosenFiniteProducts F
+  change (εIso F).symm.inv = _; congr; ext
 
 lemma μ_of_chosenFiniteProducts (X Y : C) : μ F X Y = (prodComparisonIso F X Y).inv := by
   change (μIso F X Y).symm.inv = _; congr; ext : 1; simpa using δ_of_chosenFiniteProducts F X Y
 
-attribute [local instance] OplaxMonoidal.ofChosenFiniteProducts in
 omit [F.Monoidal] in
 /-- A finite-product-preserving functor between cartesian monoidal categories is monoidal.
 
+This is the variant where `F.OplaxMonoidal` is an argument instead.
+
 This is not made an instance because it would create a diamond for the monoidal structure on
 the identity and composition of functors. -/
-noncomputable def ofChosenFiniteProducts : F.Monoidal := .ofOplaxMonoidal F
+noncomputable def ofChosenFiniteProducts'
+    (F : C ⥤ D) [PreservesFiniteProducts F] [F.OplaxMonoidal] : F.Monoidal :=
+  .ofOplaxMonoidal F
```

Comment:
Writing these intermediate constructors seems completely orthogonal to the rest of the PR, especially since you don't use them. Should this be a separate PR?